components/server/src/user/user-service.ts

```diff
@@ -236,7 +249,11 @@ export class UserService {
             } else {
                 attributionId = AttributionId.create(user);
             }
-            if (!!attributionId && (await this.hasCredits(attributionId))) {
+            if (
+                !!attributionId &&
+                (await this.hasCredits(attributionId)) &&
+                !(await this.isUnbilledTeam(attributionId))
+            ) {
```

Comment:
I found this quite complex to mentally parse, would it help if we wrote it as this?
```suggestion
            if (!!attributionId) {
              const enoughCredits = await this.hasCredits(attributionId);
              const teamWithoutBilling = await this.isUnbilledTeam(attributionId)
              if (enoughCredits && !teamWithoutBilling) {
                ...
              }
            }
            
```